### ISSUE TITLE
"Error uploading product images" when using Take a photo or Choose from device > Google Drive

### ISSUE THREAD

**Issue by designsimply** [woocommerce/woocommerce-android#3437]
Steps to reproduce:

1. Go to Products > Photos > Add photos > Take a photo.
1. While the photo is uploading tap Add Photos again and select one from the media library.
1. Wait for the upload attempt to finish.
1. Observe the error "Error uploading product images".
1. Go to Products > Photos > Add photos > Choose from device.
1. Select any photo from Google Drive.
1. Wait for the upload attempt to finish.
1. Observe the error "Error uploading product images".

**Result:** errors when uploading images using "Take a photo" or "Choose from device > Google Drive". ([3m56s](https://cloudup.com/cGb1k4dE-hD))

<img src="https://user-images.githubusercontent.com/1119271/104673663-6dfcb980-569f-11eb-9b0e-e23a869cd227.png" width="270" alt="image">

<sup>Tested with WCAndroid 5.8-rc-1 on Pixel 3 Android 11 with site testingallthetime.com WordPress 5.6 WooCommerce 4.9.0 Jetpack 9.3.1 logged in as design5279.</sup>

App logs complain about things like this (not sure if this one's relevant though):

```
[Jan-15 24:19 WP e] UTILS StackTrace: java.lang.IllegalArgumentException: Volume data not found
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:172)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
	at android.content.ContentProviderProxy.query(ContentProviderNative.java:472)
	at android.content.ContentResolver.query(ContentResolver.java:1183)
	at android.content.ContentResolver.query(ContentResolver.java:1115)
	at android.content.ContentResolver.query(ContentResolver.java:1071)
	at org.wordpress.android.util.ImageUtils.getImageOrientation(ImageUtils.java:101)
	at org.wordpress.android.util.ImageUtils.optimizeImage(ImageUtils.java:548)
	at com.woocommerce.android.media.ProductImagesUtils.getOptimizedImageUri(ProductImagesUtils.kt:169)
	at com.woocommerce.android.media.ProductImagesUtils.getOptimizedImagePath(ProductImagesUtils.kt:178)
	at com.woocommerce.android.media.ProductImagesUtils.mediaModelFromLocalUri(ProductImagesUtils.kt:50)
	at com.woocommerce.android.media.ProductImagesService.onHandleWork(ProductImagesService.kt:150)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

[Jan-15 24:19 MEDIA d] productImagesService > Dispatching request to upload content://com.google.android.apps.photos.contentprovider/4/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F55663/ORIGINAL/NONE/image%2Fjpeg/100760523
```

and this:

```
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"WPMediaPickerFragment"}
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-FETCH_MEDIA_LIST
[Jan-15 24:19 WP v] MEDIA Fetched media list for site with size: 30
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-FETCHED_MEDIA_LIST
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
```

<details>
<summary>More app logs for the timestamp matching the video, tokens removed</summary>

```
[Jan-15 24:19 WP d] API Dispatching action: UploadAction-UPLOADED_MEDIA
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-UPLOADED_MEDIA
[Jan-15 24:19 MEDIA i] MediaFilesRepository > uploaded media 233
[Jan-15 24:19 MEDIA i] productImagesService > uploaded media 233
[Jan-15 24:19 MEDIA i] productImagesService > destroyed
[Jan-15 24:19 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Jan-15 24:19 WP d] API WCTaxStore: fetchTaxClassList
[Jan-15 24:19 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Jan-15 24:19 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 UTILS i] 🔵 Tracked: product_image_settings_add_images_button_tapped, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 UTILS i] 🔵 Tracked: product_image_settings_add_images_source_tapped, Properties: {"source":"device","blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 UTILS i] 🔵 Tracked: application_closed
[Jan-15 24:19 UTILS i] 🔵 Tracked: product_image_added, Properties: {"source":"device","blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 UTILS i] 🔵 Tracked: application_opened
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"MainActivity"}
[Jan-15 24:19 WP d] API Dispatching action: WCOrderAction-FETCH_ORDERS_COUNT
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"ProductImagesFragment"}
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"MyStoreFragment"}
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"ProductListFragment"}
[Jan-15 24:19 MEDIA i] productImagesService > created
[Jan-15 24:19 MEDIA i] productImagesService > onHandleWork
[Jan-15 24:19 WP d] API Dispatching action: NotificationAction-REGISTER_DEVICE
[Jan-15 24:19 WP e] UTILS Error reading orientation of the file: /data/user/0/com.woocommerce.android/cache/wp-1597361095644.jpg - exception: Volume data not found
[Jan-15 24:19 WP e] UTILS StackTrace: java.lang.IllegalArgumentException: Volume data not found
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:172)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
	at android.content.ContentProviderProxy.query(ContentProviderNative.java:472)
	at android.content.ContentResolver.query(ContentResolver.java:1183)
	at android.content.ContentResolver.query(ContentResolver.java:1115)
	at android.content.ContentResolver.query(ContentResolver.java:1071)
	at org.wordpress.android.util.ImageUtils.getImageOrientation(ImageUtils.java:101)
	at org.wordpress.android.util.ImageUtils.optimizeImage(ImageUtils.java:548)
	at com.woocommerce.android.media.ProductImagesUtils.getOptimizedImageUri(ProductImagesUtils.kt:169)
	at com.woocommerce.android.media.ProductImagesUtils.getOptimizedImagePath(ProductImagesUtils.kt:178)
	at com.woocommerce.android.media.ProductImagesUtils.mediaModelFromLocalUri(ProductImagesUtils.kt:50)
	at com.woocommerce.android.media.ProductImagesService.onHandleWork(ProductImagesService.kt:150)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

[Jan-15 24:19 MEDIA d] productImagesService > Dispatching request to upload content://com.google.android.apps.photos.contentprovider/4/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F55663/ORIGINAL/NONE/image%2Fjpeg/100760523
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-UPLOAD_MEDIA
[Jan-15 24:19 WP d] MEDIA starting upload for: 234
[Jan-15 24:19 WP d] API Dispatching action: UploadAction-UPLOADED_MEDIA
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-UPLOADED_MEDIA
[Jan-15 24:19 WP d] API Dispatching action: NotificationAction-REGISTERED_DEVICE
[Jan-15 24:19 WP i] NOTIFS Server response OK. Device ID: 39952721
[Jan-15 24:19 UTILS i] 🔵 Tracked: product_image_settings_add_images_button_tapped, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 WP d] API Dispatching action: WCOrderAction-FETCHED_ORDERS_COUNT
[Jan-15 24:19 UTILS i] 🔵 Tracked: unfulfilled_orders_loaded, Properties: {"has_unfulfilled_orders":false,"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 UTILS i] 🔵 Tracked: product_image_settings_add_images_source_tapped, Properties: {"source":"wpmedia","blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"WPMediaPickerFragment"}
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-FETCH_MEDIA_LIST
[Jan-15 24:19 WP v] MEDIA Fetched media list for site with size: 30
[Jan-15 24:19 WP d] API Dispatching action: MediaAction-FETCHED_MEDIA_LIST
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 MEDIA w] Empty media url
[Jan-15 24:19 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"ProductImagesFragment"}
[Jan-15 24:20 WP d] MEDIA media upload successful: Response{protocol=h2, code=200, message=, url=https://public-api.wordpress.com/rest/v1.1/sites/156709371/media/new/?locale=en_GB}
[Jan-15 24:20 WP d] MEDIA mediaRestClient: removed id: 234 from current uploads, remaining: 0
[Jan-15 24:20 WP d] API Dispatching action: UploadAction-UPLOADED_MEDIA
[Jan-15 24:20 WP d] API Dispatching action: MediaAction-UPLOADED_MEDIA
[Jan-15 24:20 MEDIA i] MediaFilesRepository > uploaded media 234
[Jan-15 24:20 MEDIA i] productImagesService > uploaded media 234
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCTS
[Jan-15 24:20 WP d] API WCTaxStore: fetchTaxClassList
[Jan-15 24:20 MEDIA i] productImagesService > destroyed
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCTS
[Jan-15 24:20 UTILS i] 🔵 Tracked: product_list_loaded, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 UTILS i] 🔵 Tracked: product_image_settings_done_button_tapped, Properties: {"has_changed_data":true,"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"ProductDetailFragment"}
[Jan-15 24:20 UTILS i] 🔵 Tracked: product_detail_update_button_tapped, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-UPDATE_PRODUCT
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-UPDATED_PRODUCT
[Jan-15 24:20 UTILS i] 🔵 Tracked: product_detail_update_success, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 WP d] API WCTaxStore: fetchTaxClassList
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-FETCH_SINGLE_PRODUCT
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-FETCHED_SINGLE_PRODUCT
[Jan-15 24:20 UTILS i] 🔵 Tracked: product_detail_loaded, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-FETCH_PRODUCT_PASSWORD
[Jan-15 24:20 WP d] API WCTaxStore: fetchTaxClassList
[Jan-15 24:20 UTILS i] 🔵 Tracked: back_pressed, Properties: {"context":"MainActivity"}
[Jan-15 24:20 PRODUCTS e] CancellationException while fetching single product
[Jan-15 24:20 PRODUCTS d] can't load more products
[Jan-15 24:20 WP d] API Dispatching action: WCProductAction-FETCHED_PRODUCT_PASSWORD
[Jan-15 24:20 UTILS i] 🔵 Tracked: main_tab_dashboard_selected, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 WP d] API Dispatching action: WCStatsAction-FETCH_REVENUE_STATS
[Jan-15 24:20 WP d] API Dispatching action: WCStatsAction-FETCH_NEW_VISITOR_STATS
[Jan-15 24:20 WP d] API Dispatching action: WCOrderAction-FETCH_HAS_ORDERS
[Jan-15 24:20 UTILS i] 🔵 Tracked: dashboard_top_performers_loaded, Properties: {"range":"days","blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 WP d] API Dispatching action: WCStatsAction-FETCHED_NEW_VISITOR_STATS
[Jan-15 24:20 UTILS i] 🔵 Tracked: main_menu_settings_tapped, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"AppSettingsActivity"}
[Jan-15 24:20 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"MainSettingsFragment"}
[Jan-15 24:20 WP d] API Dispatching action: WCStatsAction-FETCHED_REVENUE_STATS
[Jan-15 24:20 UTILS i] 🔵 Tracked: dashboard_main_stats_loaded, Properties: {"range":"days","blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 WP d] API Dispatching action: WCOrderAction-FETCHED_HAS_ORDERS
[Jan-15 24:20 UTILS i] 🔵 Tracked: main_menu_contact_support_tapped, Properties: {"blog_id":156709371,"is_wpcom_store":false}
[Jan-15 24:20 UTILS i] 🔵 Tracked: view_shown, Properties: {"name":"HelpActivity"}
[Jan-15 24:20 UTILS i] 🔵 Tracked: support_application_log_viewed
[Jan-15 24:20 UTILS i] 🔵 Tracked: application_closed
[Jan-15 24:20 UTILS i] 🔵 Tracked: application_opened
[Jan-15 24:20 WP d] API Dispatching action: NotificationAction-REGISTER_DEVICE
[Jan-15 24:20 WP d] API Dispatching action: NotificationAction-REGISTERED_DEVICE
[Jan-15 24:20 WP i] NOTIFS Server response OK. Device ID: 39952721

```
</details>

**Comment by nbradbury** [comment]
Just noting that the crash is happening in `ImageUtils.getImageOrientation`, which is part of the utility library shared with WPAndroid.

**Comment by AliSoftware** [comment]
Code-freezing release/5.9 today.

As it seem this issue was planned for milestone 5.9 but hasn't been closed, it should either be closed if the bug has in fact been fixed in practice but forgot to be closed, or it will have to either be fixed directly on the release/5.9 branch very soon, or be moved to milestone 6.0.

**Comment by nbradbury** [comment]
@AliSoftware Sorry, I had started to look into this but ended up AFK due a family emergency. I've simply removed the milestone for now.
